### PR TITLE
New pipe: s337 framer

### PR DESCRIPTION
**Pull request by funman**
This pipe takes sound.s32. with s337 data and outputs complete s337 frames,
starting with the header

### CODE REVIEW

File: lib/upipe-framers/upipe_s337_framer.c [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+    if (unlikely(upipe == NULL))
+        return NULL;
+
+    struct upipe_s337f *upipe_s337f = upipe_s337f_from_upipe(upipe);
+    upipe_s337f_init_urefcount(upipe);
+    upipe_s337f_init_output(upipe);
+    upipe_s337f_init_flow_def(upipe);
+    upipe_s337f->uref = NULL;
+    upipe_throw_ready(&upipe_s337f->upipe);
+    return &upipe_s337f->upipe;
```
Comment (cmassiot) [inline lib/upipe-framers/upipe_s337_framer.c]: How about return upipe ? :)

File: lib/upipe-framers/upipe_s337_framer.c [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+        }
+        upipe_s337f_output(upipe, uref, upump_p);
+        return;
+    } else if (!output) {
+        upipe_notice(upipe, "Found synchro");
+    }
+
+    if (output) {
+        size_t size[2];
+        uref_sound_size(uref, &size[0], NULL);
```
Comment (cmassiot) [inline lib/upipe-framers/upipe_s337_framer.c]: This can fail and segfault miserably later.

Comment (funman) [reply lib/upipe-framers/upipe_s337_framer.c]: I guess I should also remove the abort() based error handling

File: lib/upipe-framers/upipe_s337_framer.c [hunk 3]
```diff
@@ -105,7 +105,7 @@ static struct upipe *upipe_s337f_alloc(struct upipe_mgr *mgr,
     upipe_s337f_init_flow_def(upipe);
     upipe_s337f->uref = NULL;
     upipe_throw_ready(&upipe_s337f->upipe);
```
Comment (cmassiot) [inline lib/upipe-framers/upipe_s337_framer.c]: same here actually :).

File: lib/upipe-framers/upipe_s337_framer.c (hunk 2 above)
Comment (cmassiot) [reply lib/upipe-framers/upipe_s337_framer.c]: yes

**Comment by cmassiot** [comment]
Can you squash it ?

**Pull request merged**